### PR TITLE
doc: elaborated on some of the crate descriptions

### PR DESCRIPTION
**Pull request by baszalmstra**

### CODE REVIEW

**Comment by baszalmstra** [comment]
[Write top level documentation for all crates](https://trello.com/c/hOGE9eOm/71-write-top-level-documentation-for-all-crates)

File: crates/mun_hir/src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,7 @@
-//! HIR provides a high-level object oriented access to Mun code.
+//! HIR provides a high-level object oriented access to Mun code. It is constructed by first parsing
```
Comment (Wodann) [inline crates/mun_hir/src/lib.rs]:
```suggestion
//! HIR provides high-level, object-oriented access to Mun code. It is constructed by first parsing
```

File: crates/mun_hir/src/lib.rs [hunk 2]
```diff
@@ -1,4 +1,7 @@
...
+//! the Mun code with the mun_syntax crate. Then it is lowered into HIR constructs, names are
```
Comment (Wodann) [inline crates/mun_hir/src/lib.rs]:
```suggestion
//! Mun code with the mun_syntax crate and then it is lowered into HIR constructs, names are
```

File: crates/mun_hir/src/lib.rs [hunk 3]
```diff
@@ -1,4 +1,7 @@
...
+//! resolved and type checking is performed. HIR is the input for both the compiler as well as the
```
Comment (Wodann) [inline crates/mun_hir/src/lib.rs]:
```suggestion
//! resolved, and type checking is performed. HIR is the input for both the compiler as well as the
```

File: crates/mun_syntax/src/lib.rs [hunk 4]
```diff
@@ -1,12 +1,11 @@
 //! Syntax Tree Library used throughout Mun.
 //!
-//! This crate is heavily inspired by Rust Analyzers ra_syntax crate.
+//! This crate is heavily inspired by Rust Analyzers ra_syntax and ra_parsing crate.
```
Comment (Wodann) [inline crates/mun_syntax/src/lib.rs]: Maybe add links to `ra_syntax` and `ra_parsing`?

**Comment by Wodann** [comment]
Looks good! 👍 
Maybe squash and then you can rebase :)

**Pull request merged**